manifests/agent/cron.pp

```diff
@@ -16,9 +17,13 @@
     $cmd = "${puppet::params::puppet_cmd} agent --confdir ${puppet::params::puppet_confdir} --onetime --no-daemonize >/dev/null"
   }
 
+  $interval = 60 / $frequency
+  $random_offset = fqdn_rand($interval)
+  $cron_schedule = $frequency.map |$value| { ($value * $interval) + $random_offset}
```

Comment:
@igalic correct, future-parser only. I'm no longer running current parser anywhere so I have no personal need for backwards compatibility, but I wanted to get feedback before going further because I realize that not everybody is ready to make the switch.

If requiring future parser isn't feasible for other puppet-puppet users, I think the second-best option would be to simply parameterize puppet::agent to accept a field passed directly to the `minute` parameter on the cron resource, defaulting to `fqdn_rand(60)`, and leaving a comment / documentation to suggest adding code like this future-parser-only material in the profile that wraps puppet-puppet for people who want to use it.

If future-parser-only code would cause problems for you, I'm totally fine solving this another way -- I just wanted to propose what I see as the most elegant fix first, in case others are fine with future parser.